Frends.LDAP.SearchObjects/Frends.LDAP.SearchObjects.Tests/UnitTests.cs

```diff
@@ -0,0 +1,559 @@
+            x.AttributeSet.Any(y => y.Key.Equals("sn")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Tuser")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("cn")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Tes Tuser")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("objectclass")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("top")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("givenname")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Te")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("title")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("engineer"))
+        ));
+
+        //Others
+        Assert.IsTrue(result.SearchResult.Any(x =>
+            x.DistinguishedName.Equals("uid=test,ou=users,dc=wimpi,dc=net") ||
+            x.DistinguishedName.Equals("CN=Qwe Rty,ou=users,dc=wimpi,dc=net") ||
+            x.DistinguishedName.Equals("CN=Foo Bar,ou=users,dc=wimpi,dc=net")));
+    }
+
+    [TestMethod]
+    public void Search_BatchSize_Test()
+    {
+        input = new()
+        {
+            SearchBase = _path,
+            Scope = Scopes.ScopeSub,
+            Filter = null,
+            MsLimit = default,
+            ServerTimeLimit = default,
+            SearchDereference = SearchDereference.DerefNever,
+            MaxResults = default,
+            BatchSize = 0,
+            TypesOnly = default,
+            Attributes = null,
+        };
+        connection = new()
+        {
+            Host = _host,
+            User = _user,
+            Password = _pw,
+            SecureSocketLayer = false,
+            Port = _port,
+            TLS = false,
+        };
+
+        var result = LDAP.SearchObjects(input, connection);
+        Assert.IsTrue(result.Success.Equals(true));
+        Assert.IsTrue(result.SearchResult.Any(x =>
+            x.DistinguishedName.Equals("CN=Tes Tuser,ou=users,dc=wimpi,dc=net") &&
+            x.AttributeSet.Any(y => y.Key.Equals("sn")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Tuser")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("cn")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Tes Tuser")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("objectclass")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("top")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("givenname")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("Te")) &&
+            x.AttributeSet.Any(y => y.Key.Equals("title")) &&
+            x.AttributeSet.Any(y => y.Value.Equals("engineer"))
+        ));
+
+        //Others
+        Assert.IsTrue(result.SearchResult.Any(x =>
+            x.DistinguishedName.Equals("uid=test,ou=users,dc=wimpi,dc=net") ||
+            x.DistinguishedName.Equals("CN=Qwe Rty,ou=users,dc=wimpi,dc=net") ||
+            x.DistinguishedName.Equals("CN=Foo Bar,ou=users,dc=wimpi,dc=net")));
+    }
+
+    [TestMethod]
+    public void Search_MaxResults_Test()
```

Comment:
This test fails when trying to run locally. For some reason all the tests go through the workflow. Could you test them by creating the docker container again and running the tests.

Reply:
Test now checks that there are only 2 results and all attributes, so there's no need to worry about which order those test users have been created.